examples/all-clusters-app/esp32/main/DeviceWithDisplay.cpp

```diff
@@ -0,0 +1,564 @@
+/*
+ *
+ *    Copyright (c) 2022 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include "DeviceWithDisplay.h"
+
+using namespace ::chip;
+using namespace ::chip::Credentials;
+using namespace ::chip::DeviceManager;
+using namespace ::chip::DeviceLayer;
+
+#if CONFIG_HAVE_DISPLAY
+static const char * TAG = "DeviceWithDisplay";
+#endif
+
+#if CONFIG_DEVICE_TYPE_M5STACK
+
+std::vector<Button> buttons          = { Button(), Button(), Button() };
+std::vector<gpio_num_t> button_gpios = { BUTTON_1_GPIO_NUM, BUTTON_2_GPIO_NUM, BUTTON_3_GPIO_NUM };
```

Comment:
Could we define a constructor function `Button::Button(gpio_num_t)` so that we don't need another vector for `gpio_num_t`?
And please use an array instead of vector.